test/apps/profiling.c

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2016.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#define HAVE_PROFILING 1
+#include <ucs/debug/profile.h>
+
+#include <stdio.h>
+
+UCS_PROFILE_FUNC_VOID(my_func, ()) {
+    UCS_PROFILE_CALL(printf, "Hello World!\n");
+}
+
```

Comment:
IMHO it is a good idea to add more profiling calls (non void func, code block) and comments. That way this example can be used as a reference.